doc/fundamentals/append_overwrite.md

```diff
@@ -0,0 +1,616 @@
+$ pachctl put-file voterData master -f OHVoterData.csv
+```
+The file will, by default, be placed into the top-level folder of
+voterData with the name "OHVoterData.csv".  If the file were 153.8KiB,
+running the command to list files in that repo would result in
+
+```
+$ pachctl list-file voterData master
+COMMIT                           NAME             TYPE COMMITTED    SIZE
+8560235e7d854eae80aa03a33f8927eb /OHVoterData.csv file 1 second ago 153.8KiB
+```
+If you were to re-run the `put-file` command, above, by default, the
+file would be appended to itself and listing the repo would look like
+this:
+```
+$ pachctl list-file voterData master
+COMMIT                           NAME             TYPE COMMITTED     SIZE
+105aab526f064b58a351fe0783686c54 /OHVoterData.csv file 2 seconds ago 307.6KiB
+```
+
+In this case, any pipelines that use this repo for input will see an updated file that has double the data in it.
+This is Pachyderm's default behavior.
+What if you want to overwrite the files?
+
+### Overwriting files
+
+This is where the `-o` (or `--overwrite`) flag comes in handy.  It will, as you've probably guessed, overwrite the file, rather than append it.
+```
+$ pachctl put-file voterData master -f OHVoterData.csv --overwrite
+$ pachctl list-file voterData master
+COMMIT                           NAME             TYPE COMMITTED    SIZE
+8560235e7d854eae80aa03a33f8927eb /OHVoterData.csv file 1 second ago 153.8KiB
+$ pachctl put-file voterData master -f OHVoterData.csv --overwrite
+$ pachctl list-file voterData master
+COMMIT                           NAME             TYPE COMMITTED    SIZE
+4876f99951cc4ea9929a6a213554ced8 /OHVoterData.csv file 1 second ago 153.8KiB
+```
+### Deduplication
+Pachyderm will deduplicate data loaded into input repositories.  If you were to load another file that hashed identically to "OHVoterData.csv", there would be one copy of the data in Pachyderm with two files' metadata pointing to it.  This works even when using the append behavior above.  If you were to put a file named OHVoterData2004.csv that was identical to that first put-file of OHVoterData.csv, and then update OHVoterData.csv as shown above, there would be two sets of bits in Pachyderm:
+
+- a set of bits that would be returned when asking for the old branch of OHVoterData.csv & OHVoterData2004.csv and
+- a set of bits that would be appended to that first set to assemble the new, master branch of OHVoterData.csv.
+
+In short, Pachyderm is smart about keeping the minimum set of bits in the object store and assembling the version of the file you (or your code!) have asked for.
+
+### Use cases for large datasets in single files
+
+#### Split and target-file flags
+This default "append" behavior is useful for file types that are often used in data science: line-delimited files and JavaScript Object Notation (json) files. `pachctl` includes the powerful `--split`, `--target-file-bytes` and `--target-file-datums` flags to deal with those files.
```

Comment:
I think this opening section needs to contextualize the problem better, not just rely on the section title. Why does split even exist? It's also missing a few details. Here's my draft of the wording.

### Splitting large files in Pachyderm

Unlike a system like Git which expects 95+% of your files to be text, Pachyderm does not do intra-file diffing because we work with any file types -- text, JSON, images, binary, etc. Pachyderm diffs content at the per-file level. Therefore, if one bit in content of a file changes, Pachyderm sees that as a new file. Similarly, Pachyderm can only distribute computation at the file-level so if your data is only one large file, it can only be processed by a single worker. 

Because of these reasons, it's pretty common to break up large files into smaller chunks. For simple data types Pachyderm provides the `--split` flag to `put-file` to automatically do this for you. For more complex splitting patterns (e.g. avro or other binary formats), you'll need to manually split your data either at ingest or with a Pachyderm pipeline. 

#### Split and target-file flags
For common file types such as JSON, text or CSV, Pachyderm includes the `--split`, `--target-file-bytes` and `--target-file-datums` flags to deal with those files.

`--split`  will divide those files into chunks based on what a "record" is. In line-delimited files, it's a line.  In json files, it's an object. Pachyderm also support splitting SQL data (specifically pgdump). `--split` takes one argument: `line`, `json`, `sql`.  This argument tells Pachyderm how you want to file delimited. For example, if you use --split `line`, Pachyderm will only divide your file on newline boundaries, never in the middle of a line. Along with the `--split` flag it's common to use additional "target" flags to get better control over the details of the split.

Reply:
There's no "sql" option in 1.8.6.  Going to leave that out.